CONTRIBUTING.md

```diff
@@ -0,0 +1,108 @@
+# Contribution Guidelines
+
+Anyone is welcome to contribute to this project. Feel free to get in touch with other community
+members on [Matrix][1], the mailing list or through issues here on GitHub.
+
+## Code of Conduct
+
+You must agree to abide by the [Mozilla Community Participation Guidelines][2]
+
+## Bug Reports
+
+You can file issues here on GitHub. Please try to include as much information as you can and under
+what conditions you saw the issue.
+
+## Sending Pull Requests
+
+Patches should be submitted as pull requests (PR).
+
+Before submitting a PR:
+
+- Ensure you are pulling from the most recent `main` branch.
+- Your code must run and pass all the automated tests before you submit your PR for review.
+  "Work in progress" or "Draft" pull requests are allowed to be submitted, but should be clearly
+  labeled as such and  should not be merged until all tests pass and the code has been reviewed.
+- Your patch should include new tests that cover your changes. It is your and your reviewer's
+  responsibility to ensure your patch includes adequate tests.
+
+When submitting a PR:
+
+- You agree to license your code under the project's open source license ([MPL 2.0](/LICENSE)).
+- Base your branch off the current `main`.
+- Add both your code and new tests if relevant.
+- Your patch must be [GPG signed][3] to ensure the commits come from a trusted source.
+- Run tests, linting and formatting checks to make sure your code complies with established 
+  standards.
+- Ensure your changes do not reduce code coverage of the test suite.
+- Please do not include merge commits in pull requests; include only commits
```

Comment:
Is this only for merge commits or merge and squash too? If we want to squash commits we should mention that too?

Reply:
"merge and squash" doesn't produce a traditional "merge commit", this guideline implies we prefer using that "squash and merge" option vs the "create merge commit" option, does that make sense?